src/code/z_eff_blure.c

```diff
@@ -66,40 +68,977 @@ void func_800A81F0(EffectBlure* this, Vec3f* p1, Vec3f* p2) {
+
+    if ((elem->flags & 3) == 2) {
+        Math_Vec3f_Diff(&sp18C, &sp1CC, &sp1B4);
+        Math_Vec3f_Diff(&sp180, &sp1C0, &sp1A8);
+    } else {
+        Vec3f sp118;
+        Vec3f sp10C;
+
+        ratio = (f32)(elem - 1)->timer / (f32)this->elemDuration;
+        EffectBlure_GetComputedValues(this, index - 1, ratio, &sp1EC, &sp1E4, &sp1DC, &sp1D8);
+        Math_Vec3s_ToVec3f(&sp118, &sp1EC);
+        Math_Vec3s_ToVec3f(&sp10C, &sp1E4);
+        Math_Vec3f_Diff(&sp18C, &sp118, &sp1B4);
+        Math_Vec3f_Diff(&sp180, &sp10C, &sp1A8);
+    }
+
+    Math_Vec3f_Scale(&sp1B4, 0.5f);
+    Math_Vec3f_Scale(&sp1A8, 0.5f);
+
+    if (((elem + 1)->flags & 3) == 2) {
+        Math_Vec3f_Diff(&sp18C, &sp1CC, &sp174);
+        Math_Vec3f_Diff(&sp180, &sp1C0, &sp168);
+    } else {
+        Vec3f sp100;
+        Vec3f spF4;
+
+        ratio = (f32)(elem + 2)->timer / (f32)this->elemDuration;
+        EffectBlure_GetComputedValues(this, index + 2, ratio, &sp1EC, &sp1E4, &sp1DC, &sp1D8);
+        Math_Vec3s_ToVec3f(&sp100, &sp1EC);
+        Math_Vec3s_ToVec3f(&spF4, &sp1E4);
+        Math_Vec3f_Diff(&sp100, &sp1CC, &sp174);
+        Math_Vec3f_Diff(&spF4, &sp1C0, &sp168);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_eff_blure/func_800A8DE8.s")
+    Math_Vec3f_Scale(&sp174, 0.5f);
+    Math_Vec3f_Scale(&sp168, 0.5f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_eff_blure/func_800A92FC.s")
+    vtx = GRAPH_ALLOC(gfxCtx, sizeof(Vtx[16]));
+    if (vtx == NULL) {
+    } else {
+        Math_Vec3f_Diff(&sp1CC, &sp138, &sp158);
+        Math_Vec3f_Scale(&sp158, 10.0f);
+        Math_Vec3f_Diff(&sp1C0, &sp138, &sp14C);
+        Math_Vec3f_Scale(&sp14C, 10.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_eff_blure/func_800A9330.s")
+        Color_RGBA8_Copy(&sp148, &sp1A4);
+        Color_RGBA8_Copy(&sp144, &sp1A0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_eff_blure/func_800A9804.s")
+        vtx[0].v = baseVtx;
+        vtx[1].v = baseVtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_eff_blure/func_800AA190.s")
+        vtx[0].v.ob[0] = func_80086814(sp158.x);
+        vtx[0].v.ob[1] = func_80086814(sp158.y);
+        vtx[0].v.ob[2] = func_80086814(sp158.z);
+        vtx[0].v.cn[0] = sp148.r;
+        vtx[0].v.cn[1] = sp148.g;
+        vtx[0].v.cn[2] = sp148.b;
+        vtx[0].v.cn[3] = sp148.a;
+        vtx[1].v.ob[0] = func_80086814(sp14C.x);
+        vtx[1].v.ob[1] = func_80086814(sp14C.y);
+        vtx[1].v.ob[2] = func_80086814(sp14C.z);
+        vtx[1].v.cn[0] = sp144.r;
+        vtx[1].v.cn[1] = sp144.g;
+        vtx[1].v.cn[2] = sp144.b;
+        vtx[1].v.cn[3] = sp144.a;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_eff_blure/func_800AA460.s")
+        for (i = 1; i < 8; i++) {
+            s32 j1 = 2 * i;
+            s32 j2 = 2 * i + 1;
+            Vec3f spE0;
+            f32 temp_f28 = i / 7.0f;                               // t
+            f32 temp_f0 = temp_f28 * temp_f28;                     // t^2
```

Comment:
```suggestion
            f32 temp_f0 = SQ(temp_f28);                     // t^2
```
May need to reformat to ensure the `// t^2` comment lines up